.github/workflows/inspektor-gadget.yml

```diff
@@ -200,66 +375,106 @@ jobs:
 
         make local-gadget-tests
 
-    - name: Lint
-      uses: golangci/golangci-lint-action@v2
-      with:
-        version: v1.43.0
-
+  test-integration:
+    name: Integration tests
+    # level: 1
+    # Add everything level 0 jobs as dependencies thus we do not run integration tests if one level 0 is failing.
+    needs: [documentation-checks, lint, test-basic, test-controller, test-local-gadget, build-kubectl-gadget, build-local-gadget, build-docker-image]
```

Comment:
I added all these dependencies because I realized it would be possible to do a release even if the linter failed and I wanted to avoid this behavior.
But I think I will remove "coding-style" checks from this, what truly matters is that code compiles, this is not something so serious if we release with non linted code.

Reply:
We can add them to the release job to be sure there is not a release if those checks don't pass. 
Also we usually do the release by tagging a commit that we're sure is passing all the CI, so this is actually not a big issue.